### PR TITLE
testSparkInterpreterDependencyLoading does not check for endToEndTestEnabled

### PR DESCRIPTION
**Pull request by prabhjyotsingh**
### What is this PR for?
This is a fix for selenium test case `testSparkInterpreterDependencyLoading` that does not check for `endToEndTestEnabled` causing build to fail on local. (`mvn $TEST_FLAG $PROFILE -B`)

Used to get this error in file `./zeppelin-server/target/failsafe-reports/org.apache.zeppelin.ZeppelinIT.txt`

    -------------------------------------------------------------------------------
    Test set: org.apache.zeppelin.ZeppelinIT
    -------------------------------------------------------------------------------
    Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.008 sec <<< FAILURE! - in org.apache.zeppelin.ZeppelinIT
    testSparkInterpreterDependencyLoading(org.apache.zeppelin.ZeppelinIT)  Time elapsed: 0.008 sec      <<< ERROR!
    java.lang.NullPointerException: null
	    at org.apache.zeppelin.ZeppelinIT.testSparkInterpreterDependencyLoading(ZeppelinIT.java:210)

### What type of PR is it?
Hot Fix

### Todos
* [x] - Fix for testSparkInterpreterDependencyLoading
* [x] - delete created notebook for cleanup.

### Is there a relevant Jira issue?
N/A

### How should this be tested?
`mvn $TEST_FLAG $PROFILE -B` should not fail on local when env variable "CI" is not set.

### CODE REVIEW

File: zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java [hunk 1]
```diff
@@ -202,46 +202,59 @@ public void testAngularDisplay() throws InterruptedException{
...
+      sparkEditBtn = pollingWait(By.xpath("//div[h3[text()[contains(.,'spark')]]]//button[contains(.,'edit')]"),
+          MAX_BROWSER_TIMEOUT_SEC);
+      sparkEditBtn.click();
+      WebElement testDepRemoveBtn = driver.findElement(By.xpath("//tr[descendant::text()[contains(.,'" +
+          artifact + "')]]/td[3]/div"));
+      sleep(5000, true);
+      testDepRemoveBtn.click();
+      driver.findElement(By.xpath("//button[contains(.,'Save')]")).submit();
+      driver.switchTo().alert().accept();
+    } catch (ElementNotVisibleException e) {
```
Comment (felixcheung) [inline zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java]: would there be any other error? like xpath returning no match (say the element has been removed)

**Comment by felixcheung** [comment]
I think it's fine keeping this in #706 - either way is ok with me.
Much easier to see diff this way: https://github.com/apache/incubator-zeppelin/pull/709/files?w=1

File: zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java (hunk 1 above)
Comment (prabhjyotsingh) [inline zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java]: Sure, what you say make sense, let me change all other catch from ElementNotVisibleException to just Exception, so, it catches every thing in https://github.com/apache/incubator-zeppelin/pull/706

**Comment by prabhjyotsingh** [comment]
Yes, agreed, this is much cleaner.

**Comment by prabhjyotsingh** [comment]
@felixcheung have replaced all "ElementNotVisibleException" to "Exception", and then log the same.

**Comment by felixcheung** [comment]
looks good.

File: zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java [hunk 2]
```diff
@@ -202,46 +202,59 @@ public void testAngularDisplay() throws InterruptedException{
...
+      WebElement testDepRemoveBtn = driver.findElement(By.xpath("//tr[descendant::text()[contains(.,'" +
+          artifact + "')]]/td[3]/div"));
+      sleep(5000, true);
+      testDepRemoveBtn.click();
+      driver.findElement(By.xpath("//button[contains(.,'Save')]")).submit();
+      driver.switchTo().alert().accept();
+    } catch (Exception e) {
+      LOG.error("Exception in ZeppelinIT while testSparkInterpreterDependencyLoading ", e);
+      File scrFile = ((TakesScreenshot) driver).getScreenshotAs(OutputType.FILE);
+
```
Comment (Leemoonsoo) [inline zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java]: Shouldn't we throw `e` after taking screenshot, if getting exception is considered as a test failure?

Comment (prabhjyotsingh) [inline zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java]: Sure did the relevant change.

Comment (felixcheung) [inline zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java]: You might want to just `throw e` - exception stack should already have the test name?

Comment (prabhjyotsingh) [inline zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java]: Thanks a lot @felixcheung, made the required change.

**Comment by prabhjyotsingh** [comment]
@felixcheung  @Leemoonsoo Thanks for the review. I have addressed the comment.

File: zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java [hunk 3]
```diff
@@ -193,55 +194,68 @@ public void testAngularDisplay() throws InterruptedException{
       sleep(100, true);
 
       System.out.println("testCreateNotebook Test executed");
-    } catch (ElementNotVisibleException e) {
+    } catch (Exception e) {
       LOG.error("Exception in ZeppelinIT while testAngularDisplay ", e);
       File scrFile = ((TakesScreenshot)driver).getScreenshotAs(OutputType.FILE);
-
+      throw new Exception(e);
```
Comment (felixcheung) [inline zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java]: can we instead say `throw e;` here? e is already an Exception
http://stackoverflow.com/questions/1097527/rethrowing-exceptions-in-java-without-losing-the-stack-trace

Comment (prabhjyotsingh) [inline zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java]: got it!!! missed it. Thank you. :)

**Comment by felixcheung** [comment]
looks good, thanks.
merging if no more comments.

**Comment by Leemoonsoo** [comment]
LGTM

**Pull request closed without merging**